src/components/registerPlugins.ts

```diff
@@ -13,30 +13,33 @@ import * as config from '../config.json'
 import DBConnect from './databaseConnect'
 import { ActionPlugin } from '../types'
 
-const pluginsFolder = path.join(getAppEntryPointDir(), config.pluginsFolder)
+const pluginsFolderFull = path.join(getAppEntryPointDir(), config.pluginsFolder)
+const pluginsFolder = config.pluginsFolder
 const pluginJsonFilename = 'plugin.json'
 const getPluginIndexPath = (pluginFolderPath: string) =>
-  fs.existsSync(path.join(pluginFolderPath, 'src', 'index.js'))
-    ? path.join(pluginFolderPath, 'src', 'index.js') // in production use index.js
-    : path.join(pluginFolderPath, 'src', 'index.ts') // otherwoise use index.ts
+  fs.existsSync(path.join(pluginsFolderFull, pluginFolderPath, 'src', 'index.js'))
+    ? path.join(pluginsFolder, pluginFolderPath, 'src', 'index.js') // in production use index.js
+    : path.join(pluginsFolder, pluginFolderPath, 'src', 'index.ts') // otherwise use index.ts
 
 export default async function registerPlugins() {
   // Load plugin info from files
   console.log('Scanning plugins folder...')
   const plugins = fs
-    .readdirSync(pluginsFolder)
-    .map((pluginFolder) => {
-      return path.join(pluginsFolder, pluginFolder)
-    })
-    .filter((pluginPath) => fs.statSync(pluginPath).isDirectory())
-    .filter((pluginPath) => fs.existsSync(path.join(pluginPath, pluginJsonFilename)))
+    .readdirSync(pluginsFolderFull)
+    .filter((pluginPath) => fs.statSync(path.join(pluginsFolderFull, pluginPath)).isDirectory())
```

Comment:
would be good to map into {pluginName, pluginFullPath, pluginJsonFilenamePath} before filtering for isDirectoy and then existence of pluginJsonFilenamePath. But that's probably stylistic, can still follow code pretty easy.

Reply:
Not entirely clear what you mean. I tried different ways of combining different path chunks into different variables but couldn't really find anything else that was simpler in all places. Can you be more specific?